Commercial skipping notifications and toggle action

b-pass opened the pull request:
An implementation to addres this feature request from the forums: http://forum.kodi.tv/showthread.php?tid=208206

This adds a new action, TOGGLE_COMMSKIP which turns Commercial Skipping on/off for the current recording only (next record starts back at the default setting from settings.xml).

Also adds an option to display notifications of commercials and their lengths.

b-pass committed:
Commercial skipping notifications and toggle action

b-pass committed:
Commercial skipping notifications and toggle action

b-pass committed:
Commercial skipping notifications and toggle action

localized all strings for commercial skip toggle and notification

fix whitespace

oops wrong shared_ptr type

Switched commskipdisable to commskipenable (makes more sense to turn something on than turn it off).  Note it still defaults to TRUE because currently it is enabled and there is no way to turn it off, this just adds a way.

Moved the settings to video/tv area and updated help a little.

typeo

move setting from settings.xml to advanced settings

put these settings in the group with the other EDL/Commercial advanced settings

b-pass committed:
localized all strings for commercial skip toggle and notification

b-pass committed:
fix whitespace

b-pass commented:
Ok I did the localizations and fixed the whitespace.  I couldn't find any documentation on what localization numbers to use, so I just found some holes.  Is that ok?

mkortstiege commented:
Please inverse the comskip option? It's easier to activate and disable instead of disable activation.

FernetMenta commented:
support for comskip is no requirement for a pvr addon. hence adding those options to general settings is not correct. this needs to be enabled by something private to an addon and conveyed through the pvr api.

ksooo commented:
I fully backup what @FernetMenta said. Currently we have a couple of general settings in the PVR area suffering from that problem. We "fix" this with the respective help text for the setting saying something like "not supported by all backends" or some other desperate words to workaround the actual design flaw. For new settings, we should do better.

ksooo commented on xbmc/ApplicationPlayer.cpp:
```diff
@@ -257,6 +257,13 @@ bool CApplicationPlayer::SeekScene(bool bPlus)
 
+void CApplicationPlayer::ToggleCommSkip()
+{
+  boost::shared_ptr<IPlayer> player = GetInternal();
```
IMO, we consequently switched to std::shared_ptr some time ago?

b-pass commented:
@ksooo, @FernetMenta CommSkip is supplied by Edit Decision Lists which can be supplied by an external (non-PVR) source as well.  PVR seems like the logical place for it.  I don't think this setting is addon-specific, but maybe it should go in videosettings instead?  Thoughts?

b-pass committed:
oops wrong shared_ptr type

b-pass committed:
Switched commskipdisable to commskipenable (makes more sense to turn something on than turn it off).  Note it still defaults to TRUE because currently it is enabled and there is no way to turn it off, this just adds a way.

Moved the settings to video/tv area and updated help a little.

b-pass committed:
typeo

ksooo commented:
The point is not about where to put the settings in the tree of Kodi general settings, but instead to put it into the settings of the addon actually supporting this feature.

The tricky part about this is, that the PVR addon API must be extended then with a function that allows Kodi to determine whether this option is available for the addon providing the current stream and that can be used by Kodi to obtain the value of the commskip setting.

Don't want to block this PR, just want to encourage people to do new things right, even if it is (much more) work. Please remember the 80/20 rule - sigh.

b-pass commented:
@ksooo But this functionality is not only provided by the addon, it existed in XBMC/Kodi before PVR, and is supported also by flat files in several different formats.  Tying the ability to enable it to a PVR addon would break that existing functionality.

MartijnKaijser commented:
in the case the commskip file is provided for normal files it's not even a PVR specific function so adding this setting in PVR section would already be pointless. 
I backup @FernetMenta and @ksooo that it should be brought through a PVR API. Yes it's more work but it the best way to do this

Montellese commented:
I have no clue about the actual feature but from the description of @b-pass it sounds like this is just a toggle switch for people who already use this feature without any PVR addon installed simply using local files. So I don't understand why everyone is talking about extending the PVR API. Why would you want to extend the PVR API for a feature that can be used without having any PVR addon installed or PVR support completely disabled?

b-pass commented:
@MartijnKaijser I moved the settings out of the PVR group and into the Video Library group to be near the other "TV" related settings.

MartijnKaijser commented on system/settings/settings.xml:
```diff
@@ -463,6 +463,16 @@
           <control type="toggle" />
         </setting>
+        <setting id="videoplayer.commskipenabled" type="boolean" label="38017" help="38018">
+          <level>1</level>
```
this in no way a lowest level setting. People who make EDL list are advanced users

ksooo commented:
@Montellese Just talking about things to my best knowledge, dude. ;-) Did not know about the non-PVR aspects of the commskip feature, but know about the PVR functionality very well. That's why I'm talking about how to IMO do it right for the PVR part. 

The non-PVR provided EDLs makes things more complicated and I have no clue how to handle this part right, especially where to put any settings for this. Obviously, there is not always a "addon-specific" settings dialog available.

My points on how to do it IMO right for PVR-provided EDLs remain unchanged, though.

MartijnKaijser commented:
The setting is not really clear that you need actual EDL files for commskip to work. This will only be a support burden because people think it will work out-of-the-box and come complaining.

ksooo commented:
@MartijnKaijser yes, I'm actually afraid of this "support nightmare", too. We have to be really very clear about what this power user feature is/is not if we expose it through any UI.

b-pass committed:
move setting from settings.xml to advanced settings

b-pass committed:
put these settings in the group with the other EDL/Commercial advanced settings

b-pass commented:
@ksooo @MartijnKaijser I moved the settings into advancedsettings.xml, there are a bunch of advanced commercial break settings for EDL already there so I think that ought to eliminate any support headaches.

ksooo commented:
Much better, imo. Thx.

mkortstiege commented:
Jep, looks better now.

MartijnKaijser commented:
@xhaggi @Jalle19 @opdenkamp and who ever. Ok to go or last comments?

rkraevskiy committed:
Mouse wheel support for X11-less linux distros (openelec)

rkraevskiy committed:
printf call replaced by CLog::Log

MartijnKaijser commented:
jenkins build this please

sialivi committed:
[DirectoryNode] add 'All albums' option

Adds the missing "All albums" option when browsing Music Videos by Artist
requested in http://forum.kodi.tv/showthread.php?tid=185443

Paxxi committed:
Got rid of boost::totally_ordered from AddonVersion

MartijnKaijser committed:
Merge pull request #7499 from uNiversaI/fonts

[DirectoryNode] add 'All albums' option

xhaggi committed:
[guilib] fix dirty regions because it evaluates always to true

Paxxi committed:
Merge pull request #7498 from Paxxi/totally_ordered

[XBMC][ADDONS] Got rid of boost::totally_ordered from AddonVersion

Paxxi committed:
Merge pull request #7478 from rkraevskiy/master

Mouse wheel support for X11-less linux distros (openelec)

ksooo committed:
Merge pull request #7501 from ryangribble/pvr_timer_showparent

[pvr] Show ".." label when displaying children of repeating timers

xhaggi committed:
Merge pull request #7504 from xhaggi/fix-dirty-regions

[guilib] fix dirty regions because it evaluates always to true

Montellese committed:
[win32] add system/ssleay32.dll to .gitignore

opdenkamp commented:
needs a squash but looks okay. but this doesn't touch anything from pvr so you're pinging the wrong ppl ;-)

b-pass committed:
Merge branch 'master' of github.com:b-pass/xbmc

b-pass commented:
Squashed.

b-pass committed:
Merge branch 'master'

his merge is necessary,

b-pass commented:
Squashed... I think?

hudokkow commented:
This PR is a mess. You're using (your) master branch to work and you shouldn't.

Please create a new branch based on master, cherry-pick your commits and open a new pull request. We can close this one after that.

b-pass commented:
Resubmitted as https://github.com/xbmc/xbmc/pull/7520
